policyengine_us/variables/gov/simulation/labor_supply_response/substitution_elasticity.py

```diff
@@ -0,0 +1,78 @@
+from policyengine_us.model_api import *
+
+
+class substitution_elasticity(Variable):
+    value_type = float
+    entity = Person
+    label = "substitution elasticity of labor supply"
+    unit = "/1"
+    definition_period = YEAR
+
+    def formula(person, period, parameters):
+        p = parameters(
+            period
+        ).gov.simulation.labor_supply_responses.elasticities.substitution
+
+        if p.all != 0:
+            return p.all
+
+        # TODO: Parametrise
+        EARNINGS_DECILE_MARKERS = [
+            0,
+            14e3,
+            28e3,
+            39e3,
+            50e3,
+            61e3,
+            76e3,
+            97e3,
+            138e3,
+            1_726e3,
+        ]
+
+        raw_earnings = add(
+            person,
+            period,
+            [
+                "employment_income_before_lsr",
+                "self_employment_income_before_lsr",
+            ],
+        )
+        earnings = max_(raw_earnings, 0)
+        earnings_decile = (
+            np.searchsorted(EARNINGS_DECILE_MARKERS, earnings) + 1
+        )
+
+        tax_unit = person.tax_unit
+        # Primary earner == highest earner in tax unit
+        max_earnings_in_unit = tax_unit.max(earnings)
+        is_primary_earner = earnings == max_earnings_in_unit
```

Comment:
Strikes me as bug-inducing

Reply:
See also https://github.com/PolicyEngine/policyengine-us/issues/4542 and https://github.com/PolicyEngine/policyengine-us/issues/5391

Reply:
Yep filed https://github.com/PolicyEngine/policyengine-us/issues/6104